src/oracle_history.ts

```diff
@@ -30,10 +30,10 @@ export class OracleHistory {
     let down: string[] = [];
 
     for (let [assetId, priceData] of current.prices) {
-      let priceHistory = this.priceHistory.get(assetId);
+      let priceHistory = this.priceHistory.get(current.oracleId + assetId);
```

Comment:
we should probably include a

`get_price(oracleId: string, assetId: string) -> PriceData | undefined`
`set...`

such that the weird key is defined within the class, and external use doesn't need to remember how the key is structured.